src/ripple/protocol/PublicKey.h

```diff
@@ -61,13 +61,41 @@ namespace ripple {
 class PublicKey
 {
 protected:
-    std::size_t size_ = 0;
-    std::uint8_t buf_[33];  // should be large enough
+    // All the constructed public keys are valid, non-empty and contain 33
+    // bytes of data.
+    static constexpr std::size_t size_ = 33;
+    std::uint8_t buf_[size_];  // should be large enough
 
 public:
     using const_iterator = std::uint8_t const*;
 
-    PublicKey() = default;
+    // This default constructed public key is used only in ValidatorList class.
+    // Publishers' master PublicKeys are used to map into their respective UNLs
+    // emptyPubKey is used to represent the keys specified in the local config
+    // file. Unlike the other published UNL, this list is not signed with any
+    // publisher's public key and hence needs to be mapped with the emptyPubKey
+    static const PublicKey emptyPubKey;
```

Comment:
This comment is not true any more. The local config is now represented by its own variable (`localPublisherList`), and isn't stored in `publisherLists_`. I don't think `emptyPubKey` is used anywhere "useful".

I say get rid of it entirely. Any place returning it as a placeholder or fallback is _wrong_, and needs to use `std::optional` instead. All the checks become unnecessary, unless they're checking for a seated `optional`.